pkg/pillar/cmd/zedrouter/server.go

```diff
@@ -682,7 +710,7 @@ func (hdl kubeConfigHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	var appInstMetaData = &types.AppInstMetaData{
 		AppInstUUID: anStatus.UUIDandVersion.UUID,
 		Data:        kubeConfig,
-		Type:        types.AppInstMetaDataTypeKubeConfig,
+		Type:        hdl.publishDataType,
 	}
 
 	publishAppInstMetadata(hdl.ctx, appInstMetaData)
```

Comment:
With this handler being used for both `kubeConfig` and `CustomStatus`, we may need to expand [AppInstMetaData.Key()](https://github.com/lf-edge/eve/blob/8.12/pkg/pillar/types/zedroutertypes.go#L3212-L3215) method to include the `Type` attribute so that one info does not overwrite the other inside pubsub. However, this may be rather rare. I'm not sure how likely it is that application publishing `kubeConfig` is also going to use `CustomStatus`.

Reply:
that's a hell of catch! thank you! I'll fix that because it doesn't matter how rare this situation is. Once it happen we gonna have very bad time debugging it.

Reply:
Now this needs to be updated: https://github.com/lf-edge/eve/blob/8.12/pkg/pillar/cmd/zedagent/handleappInstMetadata.go#L20-L35, https://github.com/lf-edge/eve/blob/8.12/pkg/pillar/cmd/zedagent/reportinfo.go#L91-L94
`appInstMetaData.Key()` is no longer only the app UUID.
Actually I don't see any reason why `PublishAppInstMetaDataToZedCloud` takes UUID and metadata type as separate args. It can read them from `appInstMetadata` arg instead.

Reply:
@milan-zededa the pattern is that when we delete an object we pass a nil appInstMetaData. We could pass a separate "delete bool" argument and pass appInstMetaData and extract the key/type from there.

But I have an API question. We defined the EVE API with a uuid string field? Is that no longer accurate? If so we should at least add a comment saying that it is not always a UUID but a unique string.

Reply:
My suggestion was only meant for pubsub, the API should remain as it is.
I assume that controller will use AppInstUUID together with Type as the unique key in the database.